tests/main/apparmor-prompting-flag-restart/task.yaml

```diff
@@ -106,9 +106,8 @@ execute: |
     fi
 
     echo "Enable prompting via snap client where possible"
-    if os.query is-core || os.query is-ubuntu-lt 22.04; then
+    if os.query is-core || os.query is-ubuntu-lt 24.04; then
```

Comment:
seems like 22.04 GCP kernel has the right bits already and this is failing?

Reply:
Well that sure is "interesting"..... and there's no debug log with kernel versions :( guess I'll have to add some debugging and see what's going on.

In general, we don't expect Jammy to have prompting. One could install a new Ubuntu kernel(/apparmor?) on Jammy (or potentially older) and get prompting to work, but I don't love the idea of testing this.

I think a more robust solution than `os.query is-ubuntu-lt 24.04` would be to check `is-ubuntu` and then check the kernel version itself. It's possible prompting could be backported to older kernels, but I hope not.

Reply:
hm debug output should contain this:
https://github.com/canonical/snapd/blob/e9b66400b1a0b8d477f8c05e3bfb66c6b119f37e/spread.yaml#L587-L588

Reply:
FWIW, I think it's ok to count 22.04 as supported, since there's a hwe kernel which may have the right patches, and given that 22.04 isn't the most recent distro around I would expect the hwe kernel to be quite popular.

Reply:
> hm debug output should contain this:
> 
> https://github.com/canonical/snapd/blob/e9b66400b1a0b8d477f8c05e3bfb66c6b119f37e/spread.yaml#L587-L588

Yeah, for some reason the debug output is not showing at all on failure?? I'm trying some things.

> FWIW, I think it's ok to count 22.04 as supported, since there's a hwe kernel which may have the right patches, and given that 22.04 isn't the most recent distro around I would expect the hwe kernel to be quite popular.

I agree, I think ideally we should change the spread checks to actually check kernel version/features for support. So instead of just asserting things about Ubuntu versions, we check whether it is Ubuntu and the kernel is [some constraints]. This might look a bit like the actual checks within snapd though, which I don't love. Ideally there's a very clear versioning scheme where prompting is only backported to kernels >= 6.x, but I'll have to check with JJ.